Utils/Dataflow/092_datasetInfoAMI/amiDatasets.py

```diff
@@ -0,0 +1,123 @@
+    sys.exit(1)
+
+ami_client = None
+PHYS_VALUES = [{"ami": "genFiltEff", "es": "gen_filt_eff"},
+              {"ami": "crossSection", "es": "cross_section"},
+              {"ami": "crossSectionRef", "es": "cross_section_ref"},
+              {"ami": "kFactor", "es": "k_factor"}]
+
+def main(argv):
+
+    stage = pyDKB.dataflow.stage.JSONProcessorStage()
+
+    stage.add_argument('--userkey', help='PEM key file', required=True)
+    stage.add_argument('--usercert', help='PEM certificate file', required=True)
+
+    exit_code = 0
+    try:
+        stage.parse_args(argv)
+        stage.process = process
+        init_ami_client(stage.ARGS.userkey, stage.ARGS.usercert)
+        stage.run()
+    except (pyDKB.dataflow.exceptions.DataflowException, RuntimeError), err:
+        if str(err):
+            str_err = str(err).replace("\n", "\n(==) ")
+            sys.stderr.write("(ERROR) %s\n" % str_err)
+        exit_code = 2
+    finally:
+        stage.stop()
+
+    sys.exit(exit_code)
+
+def init_ami_client(userkey, usercert):
+    """
+    Initialisation of AMI client into the global variable
+    :param userkey: user key pem file
+    :param usercert: user certificate pem file
+    :return:
+    """
+    global ami_client
+    try:
+        ami_client = pyAMI.client.Client('atlas', key_file=userkey, cert_file=usercert)
+        AtlasAPI.init()
+    except:
+        sys.stderr.write(
+            "Could not establish pyAMI session. Are you sure you have a valid certificate?")
+        sys.exit(1)
+
+def process(stage, message):
+
+    str = amiPhysValues(message.content())
+    stage.output(pyDKB.dataflow.messages.JSONMessage(amiPhysValues(message.content())))
+
+    return True
+
+def amiPhysValues(data):
+    """
+    Add elements in JSON string, according to theirs names in ES mapping
+    - gen_filt_eff
+    - cross_section
+    - k_factor
+    - cross_section_ref
+    """
+    dataset = data['datasetname']
+    container = remove_tid(dataset)
+    try:
+        res = ami_client.execute(['GetPhysicsParamsForDataset',
+                                  "--logicalDatasetName=%s" % container],
+                                 format='json')
+        json_str = json.loads(res)
+        for row in json_str['AMIMessage'][0]['Result'][0]['rowset'][0]['row']:
+            for field in row['field']:
+                if field['@name'] == 'paramName':
+                    for f in row['field']:
+                        if f['@name'] == 'paramValue':
+                            data[field['$']] = f['$']
+        return change_key_names(data)
+    except:
+        sys.stderr.write("Data was not found in AMI")
+
+def change_key_names(data):
+    """
+    Changing parameter names according to PHYS_VALUES dictionary.
+    :param data: JSON string
+    :return: JSON string
+    """
+    str = json.dumps(data)
+    for item in PHYS_VALUES:
+        str = str.replace("\""+item["ami"]+"\":", "\""+item["es"]+"\":")
+    return json.loads(str)
```

Comment:
It surely must work if there only AMI fields in the original JSON, as field values are mostly numeric and will hardly interfire with names. But I\`m not sure about the rest of the document. Can\`t something similar appear in the dataset name?
I would suggest something like
```
for item in PHYS_VALUES:
    if item["ami"] in data:
        data[item["es"]] = data.pop(item["ami"])
```

And the same as above about variable names. Possible misundestanding can also look like this:
```
>>> str.replace
<method 'replace' of 'str' objects>
>>> str.replace('aac', 'a', 'b')
'bbc'
>>> str = 'aac'
>>> str.replace
<built-in method replace of str object at 0x7ff56c8d7530>
>>> str.replace('aac', 'a', 'b')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: an integer is required
```
The one who see `str.replace` automatically thinks about first variant, while actually it is the second.